pandas/core/arrays/period.py

```diff
@@ -83,88 +103,216 @@ def wrapper(self, other):
     return compat.set_function_name(wrapper, opname, cls)
 
 
-class PeriodArrayMixin(DatetimeLikeArrayMixin):
-    @property
-    def _box_func(self):
-        return lambda x: Period._from_ordinal(ordinal=x, freq=self.freq)
+class PeriodArray(DatetimeLikeArrayMixin, ExtensionArray):
+    """
+    Pandas ExtensionArray for storing Period data.
 
-    @cache_readonly
-    def dtype(self):
-        return PeriodDtype.construct_from_string(self.freq)
+    Users should use the :func:`period_array` function to create
```

Comment:
The various `*_range` function will return a `PeriodIndex` (Intervalndex, DatetimeIndex)`. Does it make sense to add a keyword to get the array instead of an index?

Reply:
Ah, yes of course. I don't think it is a big short term problem that the `*_range` functions create index objects (if passed to eg Series, they do the correct thing), so let's leave that for a future issue.